src/functions-and-arrays.js

```diff
@@ -1,41 +1,112 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(number1, number2) {
+
+  if (number1 === number2) {
+    return number1
+  } else if (number2 > number1) {
+    return number2
+  } else {
+    return number1
+  }
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  let longest
+
+  if (words.length === 0) {
+    return null
+  }
+  for (let i = 0; i < words.length; i++) {
+    const word = words[i]
+
+    if (longest === undefined) {
+      longest = word 
+    } else if (word.length > longest.length){
+      longest = word
+    }
+  }
+  return longest
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0
+
+  for (let i = 0; i < numbers.length; i++) {
+    const number = numbers[i]
+
+    if (sum === undefined) {
+      sum = number
+    } else {
+      sum += number
+    }
+  }
+  return sum
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+
+function sum(numbers) {
+  
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+
+  if (numbersAvg.length === 0) {
+    return null
+  }
+  return sumNumbers(numbersAvg) / numbersAvg.length
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+
+  if (wordsArr.length === 0) {
+    return null
+  }
+
+let sum = 0
+  
+for (let i = 0; i < wordsArr.length; i++) {
+    const wordlength = wordsArr[i].length
+
+    if (sum === undefined) {
+      sum = wordlength
+    } else {
+      sum += wordlength
+    }
+  }
+  return sum / wordsArr.length
+}
```

Comment:
Aqui nos ocurre lo mismo que antes :D
```suggestion
  for (let i = 0; i < wordsArr.length; i++) {
    sum += wordsArr[i].length
  }
  return sum / wordsArr.length
}
```